src/Microsoft.IdentityModel.Protocols/Configuration/IConfigurationRetriever.cs

```diff
@@ -30,8 +30,18 @@
 
 namespace Microsoft.IdentityModel.Protocols
 {
+    /// <summary>
+    /// Interface that defines methods to retriever populated configuration.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
     public interface IConfigurationRetriever<T>
     {
+        /// <summary>
+        /// Retrieves a populated configuration given an address and an <see cref="IDocumentRetriever"/>.
+        /// </summary>
+        /// <param name="address">address of the discovery document.</param>
+        /// <param name="retriever">the <see cref="IDocumentRetriever"/> to use to read the discovery document.</param>
+        /// <param name="cancel"><see cref="CancellationToken"/>.</param>
```

Comment:
Should read something like "A <cancellationToken> that can be used to cancel the asynchronous call." Please also capitalize.